src/main/java/org/takes/http/BkBasic.java

```diff
@@ -71,18 +71,29 @@ public BkBasic(final Take tks) {
         this.take = tks;
     }
 
+    @SuppressWarnings ("PMD.AvoidInstantiatingObjectsInLoops")
     @Override
     public void accept(final Socket socket) throws IOException {
         final InputStream input = socket.getInputStream();
+        final BufferedOutputStream output = new BufferedOutputStream(
+            socket.getOutputStream()
+        );
         try {
-            this.print(
-                BkBasic.addSocketHeaders(
-                    new RqLive(input),
-                    socket
-                ),
-                new BufferedOutputStream(socket.getOutputStream())
-            );
+            while (true) {
+                this.print(
+                    BkBasic.addSocketHeaders(
+                        new RqLive(input),
+                        socket
+                    ),
+                    output
+                );
+                if (input.available() <= 0) {
+                    break;
+                }
+            }
         } finally {
+            output.flush();
```

Comment:
@sebing No need to call `flush()` here, since `BufferedOutputStream` is an inheritor of `FilterOutputStream` it will be flushed upon close.